### PR TITLE
87 delete product

### PR DESCRIPTION
**Pull request by garrett-vangilder**
## Status

**READY
## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.
## Description

A few sentences describing the overall goals of the pull request's commits.

This PR allows for users to delete an item if the user is the associated user to the item.
## Resolves Issue Number
#87
## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

``` sh
git pull --prune
git checkout 87-delete-product
dotnet run
```
## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
1. View index page
2. Select delete button next to item (red X)
3. You will be rerouted to the product type page associated with the deleted item.
## Impacted Files in Application

List general components of the application that this PR will affect:
- Product Controller and Product Type View

### CODE REVIEW

**Comment by matthamil** [comment]
My `X` isn't red :( @garrett-vangilder 

LGTM :shipit:

**Review by matthamil — APPROVED** [review]
LGTM

**Comment by daynewright** [comment]
LGTM

File: Controllers/ProductsController.cs [hunk 1]
```diff
@@ -165,24 +169,53 @@ public IActionResult New()
...
 
             return View(product);
 
         }
+
+        public async Task<IActionResult> Delete([FromRoute]int id)
+        {
+            Product originalProduct = await context.Product.SingleAsync(p => p.ProductId == id);
+
+            if (!ModelState.IsValid)
```
Comment (MattKraatz) [inline Controllers/ProductsController.cs]: This shouldn't check for ModelState but instead the return of our Linq statement:
`if (originalProduct == null)`

**Review by MattKraatz — CHANGES_REQUESTED** [review]
Just one small change on my end

File: Controllers/ProductsController.cs (hunk 1 above)
Comment (garrett-vangilder) [reply Controllers/ProductsController.cs]: Good call!

**Comment by Anulfo** [comment]
LGTM

**Review by MattKraatz — APPROVED** [review]

**Comment by MattKraatz** [comment]
LGTM

**Pull request merged**